lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+        self.precip_cube.data[0, 0, 7, 7] = 1.
+        # Set first-guess data zero point that will be increased
+        self.fg_cube.data[0, 7, 7] = 0.
+        # No halo - we're only testing this method.
+        plugin = Plugin(0.)
+        expected = set_up_cube_with_no_realizations(zero_point_indices=[])
+        # expected.data contains all ones.
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_lrate_halo(self):
+        """Test that zero lightning rates increase lightning risk"""
+        # Set prob(precip) data for lowest threshold to to 1., so it has a Null
+        # impact when lightning is present.
+        self.precip_cube.data[0, 0, 7, 7] = 1.
+        # Set lightning data to zero to represent the data halo
+        self.ltng_cube.data[0, 7, 7] = 0.
+        # Set first-guess data zero point that will be increased
+        self.fg_cube.data[0, 7, 7] = 0.
+        # No halo - we're only testing this method.
+        plugin = Plugin(0.)
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.25
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+
+class Test_apply_precip(IrisTest):
+
+    """Test the apply_precip method."""
+
+    def setUp(self):
+        """Create cubes with a single zero prob(precip) point.
+        The cubes look like this:
+     precipitation_amount / (kg m^-2)
+     Dimension coordinates:
+        time: 1;
+        projection_y_coordinate: 16;
+        projection_x_coordinate: 16;
+     Auxiliary coordinates:
+          forecast_period (on time coord): 4.0 hours (simulates UM data)
+     Scalar coordinates:
+          forecast_reference_time: 2015-11-23 03:00:00
+     Data:
+       self.fg_cube:
+          forecast_period (on time coord): 0.0 hours (simulates nowcast data)
+          All points contain float(1.)
+          Cube name is "probability_of_lightning".
+       self.precip_cube:
+          With extra coordinate of length(3) "threshold" containing
+          points [0.5, 7., 35.] mm hr-1.
+          All points contain float(1.) except the
+          zero point [0, 0, 7, 7] which is float(0.)
+          and [1:, 0, ...] which are float(0.)
+          Cube name is "probability_of_precipitation".
+          Cube has added attribute {'relative_to_threshold': 'above'}
```

Comment:
So zero probability of high or extreme precip everywhere, 100% probability of light precip everywhere except at the zero point?

Reply:
Yes